extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -0,0 +1,59 @@
+import { Uri } from 'vscode';
+import * as yaml from 'js-yaml';
+import * as fs from 'fs-extra';
+import { showAndLogErrorMessage, showAndLogInformationMessage } from './helpers';
+import { Credentials } from './authentication';
+
+interface Config {
+  repositories: string[];
+  ref?: string;
+  language: string;
+}
+
+export default async function runRemoteQuery(credentials: Credentials, uri?: Uri) {
+  if (!uri || !uri.fsPath.endsWith('.ql')) {
+    return;
+  }
+
+  const octokit = await credentials.getOctokit();
+  const token = await credentials.getToken();
+
+  const queryFile = uri.fsPath;
+  const query = await fs.readFile(queryFile, 'utf8');
+
+  const repositoriesFile = queryFile.substring(0, queryFile.length - '.ql'.length) + '.repositories';
+  if (!(await fs.pathExists(repositoriesFile))) {
+    void showAndLogErrorMessage(`Missing file: '${repositoriesFile}' to specify the repositories to run against.`);
```

Comment:
Maybe add some more context? Something like:

```suggestion
    void showAndLogErrorMessage(`Missing file: '${repositoriesFile}' to specify the repositories to run against. This file must be a sibling of ${queryFile}.`);
```